From 6b7afd0fbd9777d8d13d10c330062edf5e8baaba Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 19 Jul 2022 13:15:00 +0100
Subject: [PATCH] dtoverlays: Add nohdmi options to vc4-kms-v3d overlays

There are cases where hotplug detect is incorrectly wired
or it is desired that the HDMI outputs are disabled.

Add an override to vc4-kms-v3d and vc4-kms-v3d-pi4 to do that.

https://forums.raspberrypi.com/viewtopic.php?t=337623

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README                      | 5 +++++
 arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts     | 1 +
 arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts | 3 +++
 3 files changed, 9 insertions(+)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 66219591c867..d9301e206e3d 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -4126,6 +4126,7 @@ Params: cma-512                 CMA is 512MB (needs 1GB)
         noaudio                 Disable all HDMI audio (default "off")
         composite               Enable the composite output (default "off")
                                 N.B. Disables all other outputs on a Pi 4.
+        nohdmi                  Disable HDMI output
 
 
 Name:   vc4-kms-v3d-pi4
@@ -4149,6 +4150,10 @@ Params: cma-512                 CMA is 512MB
         noaudio                 Disable all HDMI audio (default "off")
         composite               Enable the composite output (disables all other
                                 outputs)
+        nohdmi                  Disable both HDMI 0 & 1 outputs
+        nohdmi0                 Disable HDMI 0 output
+        nohdmi1                 Disable HDMI 1 output
+
 
 
 Name:   vc4-kms-vga666
diff --git a/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts b/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
index 351fc160e803..1e9afd84a85c 100644
--- a/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
@@ -119,5 +119,6 @@ __overrides__ {
 		audio   = <0>,"!13";
 		noaudio = <0>,"=13";
 		composite = <0>, "=11";
+		nohdmi = <0>, "-1-7";
 	};
 };
diff --git a/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts b/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts
index 76229cad7803..39df2133b3ee 100644
--- a/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-v3d-pi4-overlay.dts
@@ -193,5 +193,8 @@ __overrides__ {
 			    <0>, "!16",
 			    <0>, "=21",
 			    <0>, "=22";
+		nohdmi0 =   <0>, "-1-3-8";
+		nohdmi1 =   <0>, "-2-4-10";
+		nohdmi =    <0>, "-1-2-3-4-8-10";
 	};
 };
-- 
2.32.0

